[ZT] Secure your Internet traffic and SaaS apps (#12460)
cloudflare/cloudflare-docs: content/learning-paths/secure-internet-traffic/secure-saas-applications/layer-security.md

---
title: Layer security methods
pcx_content_type: learning-unit
weight: 2
layout: learning-unit
---

Once you have decided on a model (or mixture of models) to secure front door access to your SaaS applications, you can deliver a number of Cloudflare controls that address other common SaaS security areas of concern:

> - Device posture controls via either Access for SaaS or via controls related to WARP enrollment. Device posture can determine what constitutes a managed endpoint accessing a SaaS application using provisioning or access to a dedicated egress IP address.

Reviewer: ```suggestion
- Device posture controls via either Access for SaaS or via controls related to WARP enrollment. Device posture can determine what constitutes a managed endpoint accessing a SaaS application.
```

> - Device posture controls via either Access for SaaS or via controls related to WARP enrollment. Device posture can determine what constitutes a managed endpoint accessing a SaaS application using provisioning or access to a dedicated egress IP address.

Reviewer: Opnion: Perhaps these bullets could have their own sections, e.g. 
## Device posture controls
## Inline security

>   - You can also define what constitutes a healthy or semi-healthy managed endpoint and invoke step-up security policies based on potential security concerns. These controls are separate from the controls enacted for unmanaged endpoints.

Reviewer: ```suggestion
You can also define what constitutes a healthy or semi-healthy managed endpoint and invoke step-up security policies based on potential security concerns. These controls are separate from the controls enacted for unmanaged endpoints.
```

Reviewer: Opinion (feel free to ignore): I'd avoid using bullet points if there's only one item in the list.

> - For unmanaged endpoints, you can deliver selective inline security like upload/download controls, HTTP filtering, and data loss detection/prevention policies in SaaS applications without the need to install agents.

Reviewer: Should this talk about inline security options for both managed endpoints and unmanaged endpoints?

>   - You can be deliver this method through Browser Isolation, but it requires a [front door control](/learning-paths/secure-internet-traffic/secure-saas-applications/sso-front-door/) (such as Access for SaaS, dedicated egress IP for SSO, or Okta second factor) in order to be applied consistently.

Reviewer: ```suggestion
You can deliver inline security through Browser Isolation, but it requires a [front door control](/learning-paths/secure-internet-traffic/secure-saas-applications/sso-front-door/) (such as Access for SaaS) in order to be applied consistently.
```

## Browser Isolation

Cloudflare can deliver Browser Isolation in three different ways:

1. Via Cloudflare's endpoint agent (WARP), where it appears transparent. In this case, a user's browser will display `example.com`, but Cloudflare will isolate the traffic.

> 2. Clientless Web Isolation via static links, where Cloudflare appends to the web address. In this case, a user's browser will display `example.cloudflareaccess.com/browser/organization.application.com`.

Reviewer: ```suggestion
2. Clientless Web Isolation via static links, where Cloudflare appends to the web address. In this case, a user's browser will display `<your-team-name>.cloudflareaccess.com/browser/https://www.example.com`.
```

> 3. Clientless Web Isolation via Access applications, where it appears transparent. In this case, the Cloudflare public hostname will be in the Access policy for `access.example.com`, and Cloudflare isolates the traffic.

Reviewer: Afaik point 3 is only available for self-hosted apps connected via Cloudflare Tunnel:
https://developers.cloudflare.com/cloudflare-one/policies/access/isolate-application/

How does this apply to securing SaaS apps?
